--- origsrc/x264-snapshot-20090301-2245/Makefile	2009-03-01 15:45:20.000000000 -0600
+++ src/x264-snapshot-20090301-2245/Makefile	2009-03-02 10:50:07.468750000 -0600
@@ -151,7 +151,7 @@ install: x264$(EXE) $(SONAME)
 	install -m 644 x264.pc $(DESTDIR)$(libdir)/pkgconfig
 	install x264$(EXE) $(DESTDIR)$(bindir)
 	$(RANLIB) $(DESTDIR)$(libdir)/libx264.a
-ifeq ($(SYS),MINGW)
+ifneq ($(filter CYGWIN MINGW, $(SYS)), '')
 	$(if $(SONAME), install -m 755 $(SONAME) $(DESTDIR)$(bindir))
 else
 	$(if $(SONAME), ln -sf $(SONAME) $(DESTDIR)$(libdir)/libx264.$(SOSUFFIX))
--- origsrc/x264-snapshot-20090301-2245/configure	2009-03-01 15:45:20.000000000 -0600
+++ src/x264-snapshot-20090301-2245/configure	2009-03-02 10:53:04.765625000 -0600
@@ -208,13 +208,8 @@ case $host_os in
     LDFLAGS="$LDFLAGS -lm"
     ;;
   cygwin*)
-    SYS="MINGW"
+    SYS="CYGWIN"
     EXE=".exe"
-    DEVNULL="NUL"
-    if cc_check "" -mno-cygwin; then
-        CFLAGS="$CFLAGS -mno-cygwin"
-        LDFLAGS="$LDFLAGS -mno-cygwin"
-    fi
     ;;
   mingw*)
     SYS="MINGW"
@@ -242,7 +237,7 @@ case $host_cpu in
     fi
     if [ "$SYS" = MACOSX ]; then
       ASFLAGS="$ASFLAGS -f macho -DPREFIX"
-    elif [ "$SYS" = MINGW ]; then
+    elif [ "$SYS" = MINGW ] || [ "$SYS" = CYGWIN ]; then
       ASFLAGS="$ASFLAGS -f win32 -DPREFIX"
     else
       ASFLAGS="$ASFLAGS -f elf"
@@ -463,6 +458,10 @@ if [ "$shared" = "yes" ]; then
         echo "SONAME=libx264-$API.dll" >> config.mak
         echo 'IMPLIBNAME=libx264.dll.a' >> config.mak
         echo 'SOFLAGS=-Wl,--out-implib,$(IMPLIBNAME) -Wl,--enable-auto-image-base' >> config.mak
+    elif [ "$SYS" = "CYGWIN" ]; then
+        echo "SONAME=cygx264-$API.dll" >> config.mak
+        echo 'IMPLIBNAME=libx264.dll.a' >> config.mak
+        echo 'SOFLAGS=-Wl,--out-implib,$(IMPLIBNAME) -Wl,--enable-auto-image-base' >> config.mak
     elif [ "$SYS" = "MACOSX" ]; then
         echo "SOSUFFIX=dylib" >> config.mak
         echo "SONAME=libx264.$API.dylib" >> config.mak
