--- origsrc/x264-snapshot-20170330-2245-stable/configure	2017-03-30 15:45:06.000000000 -0500
+++ src/x264-snapshot-20170330-2245-stable/configure	2017-03-31 02:18:57.729883100 -0500
@@ -721,8 +721,6 @@ case $host_cpu in
                 cc_check "" "-S" && grep -q "_main:" conftest && ASFLAGS="$ASFLAGS -DPREFIX"
                 cc_check "" "-Wl,--high-entropy-va" && LDFLAGS="$LDFLAGS -Wl,--high-entropy-va"
                 LDFLAGS="$LDFLAGS -Wl,--dynamicbase,--nxcompat,--tsaware"
-                LDFLAGSCLI="$LDFLAGSCLI -Wl,--image-base,0x140000000"
-                SOFLAGS="$SOFLAGS -Wl,--image-base,0x180000000"
                 RCFLAGS="--target=pe-x86-64 $RCFLAGS"
             fi
         else
@@ -1420,7 +1418,7 @@ fi
 
 if [ "$shared" = "yes" ]; then
     API=$(grep '#define X264_BUILD' < ${SRCPATH}/x264.h | cut -f 3 -d ' ')
-    if [ "$SYS" = "WINDOWS" -o "$SYS" = "CYGWIN" ]; then
+    if [ "$SYS" = "WINDOWS" ]; then
         echo "SONAME=libx264-$API.dll" >> config.mak
         if [ $compiler_style = MS ]; then
             echo 'IMPLIBNAME=libx264.dll.lib' >> config.mak
@@ -1436,6 +1434,10 @@ if [ "$shared" = "yes" ]; then
             echo 'IMPLIBNAME=libx264.dll.a' >> config.mak
             echo "SOFLAGS=-shared -Wl,--out-implib,\$(IMPLIBNAME) $SOFLAGS" >> config.mak
         fi
+    elif [ "$SYS" = "CYGWIN" ]; then
+        echo "SONAME=cygx264-$API.dll" >> config.mak
+        echo "IMPLIBNAME=libx264.dll.a" >> config.mak
+        echo "SOFLAGS=-shared -Wl,--out-implib,\$(IMPLIBNAME) $SOFLAGS" >> config.mak
     elif [ "$SYS" = "MACOSX" ]; then
         echo "SOSUFFIX=dylib" >> config.mak
         echo "SONAME=libx264.$API.dylib" >> config.mak
--- origsrc/x264-snapshot-20170330-2245-stable/input/avs.c	2017-03-30 15:45:06.000000000 -0500
+++ src/x264-snapshot-20170330-2245-stable/input/avs.c	2017-03-31 02:16:36.564522500 -0500
@@ -46,6 +46,7 @@
 #if USE_AVXSYNTH
 #include "extras/avxsynth_c.h"
 #else
+#include <windows.h>
 #include "extras/avisynth_c.h"
 #endif
 #define AVSC_DECLARE_FUNC(name) name##_func name
